test/AST-Quake/array.cpp

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: cudaq-quake %s | cudaq-opt | FileCheck %s
+
+#include <cudaq.h>
+
+struct T {
+   void operator()(int N) __qpu__ {
+      cudaq::qreg Q(N);
+      x(Q);
+   }
+};
+
+struct S {
+   void operator()() __qpu__ {
+      int arr[3] = {4, 8, 10};
```

Comment:
Will this also work for `int arr[] = {4, 8, 10};`?
And what about `int arr[3] = {4}; // Normal C/C++ does implicit initialization of the remaining elements to 0`?

Reply:
The first is fine. I need to do a bit more to get the zero padding cases working.

Reply:
> The first is fine. I need to do a bit more to get the zero padding cases working.

OK. Will the zero padding cases be worked under this PR, or should we just open a separate issue to track it?

Reply:
I'll continue to work on it. I don't think we need a new issue. (Would partially duplicate #490.)